src/view.tree.js

```diff
@@ -23,13 +23,17 @@ class TreeView {
 
     $jstree.settings.core.data = (node, cb) => {
       const prMode = this.store.get(STORE.PR) && repo.pullNumber;
-      const loadAll = this.adapter.canLoadEntireTree() && (prMode || this.store.get(STORE.LOADALL));
+      const loadAll = this.adapter.canLoadEntireTree(repo) && (prMode || this.store.get(STORE.LOADALL));
 
       node = !loadAll && (node.id === '#' ? {path: ''} : node.original);
 
       this.adapter.loadCodeTree({repo, token, node}, (err, treeData) => {
         if (err) {
-          $(this).trigger(EVENT.FETCH_ERROR, [err]);
+          if (err.status === 206 && loadAll) { // The repo is too big to load all, need to retry
+            $jstree.refresh(true);
```

Comment:
What does this do? Refresh lazily?

Reply:
It checks wether the tree was too big , if yes it performs a reload instead of showing an error message. Since the number 206 refers to the HTTP status code, I felt like it will be odd to name it somehow through a constant. But still I agree that it is kind of ugly.